### DISCUSSION TITLE
How to find the best step?

### DISCUSSION THREAD

**Discussion by jeremy-linchao** [d8ahazard/sd_dreambooth_extension#1128] [Q&A]
When train a human face, I want to find the best step, but  I have no accurate way to do that , is there any good idea?

What I do now:
I use a small save step, such as 5, train and save a lot, and also I add a sanity prompt such as "photo of sks man by Tomer Hanuka" suggested by the youtuber  refered in the main page, and I can know which step is overfit if the sample has not anime sytle. But to find the best step, I need to use prompt scripts and iterate the checkpoint name and generate image to see which is best. I think it is too tedious

Question:
Is there any method to quickly find out the best step and can stop training by math, for example throgh the loss plot ?

**Comment by sailxjx** [comment]
I use this process:
1. Generate 4 pictures in a batch with one checkpoint, save the grid picture.
2. Try different prompts and random seeds, save the grid picture.
3. Then you have some picture grids, try other checkpoints with the prompts from these grids.
4. Compare different pictures generated by different checkpoints.

It's still a very manual and hard job, so I wonder if there are any tools that can automate this process, and also comparing the pictures is not an easy job, it's hard to say which is better, some of the pictures generated by the earlier steps may not be the same the face of the source image, but the shadows, lighting and generalization are better, and the pictures generated in the later steps may look more like people, but the generalization ability is poor, and the background, clothes, hairstyles, etc. can hardly be changed.
